crates/bevy_ui/src/layout/mod.rs

```diff
@@ -5,44 +5,69 @@ use crate::{ContentSize, Node, Style, UiScale};
 use bevy_ecs::{
     change_detection::DetectChanges,
     entity::Entity,
-    event::EventReader,
+    prelude::Component,
     query::{With, Without},
+    reflect::ReflectComponent,
     removal_detection::RemovedComponents,
-    system::{Query, Res, ResMut, Resource},
+    system::{ParamSet, Query, Res, ResMut, Resource},
     world::Ref,
 };
 use bevy_hierarchy::{Children, Parent};
 use bevy_log::warn;
 use bevy_math::Vec2;
+use bevy_reflect::{std_traits::ReflectDefault, Reflect};
 use bevy_transform::components::Transform;
 use bevy_utils::HashMap;
-use bevy_window::{PrimaryWindow, Window, WindowResolution, WindowScaleFactorChanged};
+use bevy_window::Window;
 use std::fmt;
 use taffy::{prelude::Size, style_helpers::TaffyMaxContent, Taffy};
 
+/// Marks an entity as `UI root entity` with an associated root Taffy node and holds the resolution and scale factor information necessary to compute a UI layout.
+#[derive(Component, Debug, Reflect)]
+#[reflect(Component, Default)]
 pub struct LayoutContext {
-    pub scale_factor: f64,
-    pub physical_size: Vec2,
-    pub min_size: f32,
-    pub max_size: f32,
+    /// The size of the root node in the layout tree.
+    ///
+    /// Should match the size of the output window in physical pixels of the display device.
+    pub root_node_size: Vec2,
+    /// [`Style`] properties of UI node entites with `Val::Px` values are multiplied by the `combined_scale_factor` before they are copied to the Taffy layout tree.
+    ///
+    /// `combined_scale_factor` is calculated by multiplying together the `scale_factor` of the output window and [`crate::UiScale`].
+    pub combined_scale_factor: f64,
+    /// After a UI layout has been computed, the layout coordinates are multiplied by `layout_to_logical_factor` to determine the final size of each UI Node entity to be stored in its [`Node`] component.
+    ///
+    /// `layout_to_logical_factor` is the reciprocal of the target window's `scale_factor` and doesn't include [`crate::UiScale`].
+    pub layout_to_logical_factor: f64,
 }
 
-impl LayoutContext {
-    /// create new a [`LayoutContext`] from the window's physical size and scale factor
-    fn new(scale_factor: f64, physical_size: Vec2) -> Self {
+impl Default for LayoutContext {
```

Comment:
I'm a bit skeptical about this change. The `new()` was a lot more clear to me. Creating an implicit 800x600 layout feels very arbitrary (and undocumented).

Reply:
The default values will be overwritten by `ui_layout_system` with the window's values if running in a window. 

The default values are just there for convenience for the tests. You are right though, it might be confusing and seen as suggestive of something else. I'll just derive default.